### DISCUSSION TITLE
List all installed tools and their licenses

### DISCUSSION THREAD

**Discussion by AlexanderBartholomaeus** [metagenome-atlas/atlas#576] [General]
Is there a way to list all used tools and maybe also their licenses? Your tools is very nice but I could not find a dependency file nor I was able to list the tools in conda.

**Comment by SilasK** [comment]
I don't know that there is a easy way, unfortunately. I asked others on [Twitter](https://twitter.com/SilasKieser/status/1588097775568183296).

This is the place with all the conda files with the main dependencies. https://github.com/metagenome-atlas/atlas/tree/master/atlas/workflow/envs However each of these tools depends on other software. 

I would love ther to be an easy way to list all the software and their version and license. 

I expect to be all tools to be usabely in academia and industry except eggNOG mapper.

**Reply by AlexanderBartholomaeus** [reply]
Can we trace back the single environments of the single tools? They must be accessible in the atlas env, right? Maybe I can write a small script to do that

**Comment by SilasK** [comment]
You asked for the license of the tools, isn't it? 
In theory, you can check all the files listed in the envs folder and check the licenses of the mentioned tools. 

There may be a way to automate this by using the dependency graph in GitHub. Do you know a way? 

It is true not all environments are actually installed. The installed environments are in the `database_dir/conda_envs` where database_dir is defined during atlas init. 

One could also parse the environment files for the version.
